### ISSUE TITLE
I/O operation on closed file

### ISSUE THREAD

**Issue by zohozer** [yt-dlp/yt-dlp#2837]
### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.02.04**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

I am using the 2022.2.4 version of `yt-dlp` on a Windows10 machine and I do want to download the following video file [url=https://youtu.be/dlBfVitn-ow][/url] and every time I do try I get the following error message:

`I/O operation on closed file`

or, if I remove my `config.txt` file fot `yt-dlp`, I do get this error message:

`ERROR: Did not get any data blocks`

Searching for a solution on internet I found this short clip on YT explaining for a possible solution:

[url=https://youtu.be/onaJ5SWSjDI]PYTHON : ValueError : I/O operation on closed file(url)

Please help to fix this issue.

### Verbose log

```shell
>yt-dlp.exe https://youtu.be/dlBfVitn-ow

[youtube] dlBfVitn-ow: Downloading webpage
[youtube] dlBfVitn-ow: Downloading android player API JSON
[info] dlBfVitn-ow: Downloading 1 format(s): 22
[info] Downloading video thumbnail 41 ...
[info] Writing video thumbnail 41 to: Top 10 Richest Rockstars Of All Time.webp
[download] Destination: Top 10 Richest Rockstars Of All Time.mp4
[download]   0.8% of 39.77MiB at 500.71KiB/s ETA 01:20ERROR: I/O operation on closed file
E:\temp>yt-dlp.exe -o rock.mp4 https://youtu.be/dlBfVitn-ow

[youtube] dlBfVitn-ow: Downloading webpage
[youtube] dlBfVitn-ow: Downloading android player API JSON
[info] dlBfVitn-ow: Downloading 1 format(s): 22
[info] Downloading video thumbnail 41 ...
[info] Writing video thumbnail 41 to: rock.webp
[download] Destination: rock.mp4
[download]   0.8% of 39.77MiB at 504.80KiB/s ETA 01:19ERROR: I/O operation on closed file
```

**Comment by coletdjnz** [comment]
Please follow the template and make sure you enable verbose log next time. But since I can reproduce I'll add it this time.

Fresh download:
```
$ yt-dlp https://youtu.be/dlBfVitn-ow  -f 22 -v --force-over
[debug] Command-line config: ['https://youtu.be/dlBfVitn-ow', '-f', '22', '-v', '--force-over']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.02.04 [c1653e9ef]
[debug] Lazy loading extractors is disabled
[debug] Python version 3.10.2 (CPython 64bit) - Linux-5.15.21-1-MANJARO-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4.1 (setts), ffprobe 4.4.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, secretstorage, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: https://youtu.be/dlBfVitn-ow
[youtube] dlBfVitn-ow: Downloading webpage
[youtube] dlBfVitn-ow: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[info] dlBfVitn-ow: Downloading 1 format(s): 22
[debug] Invoking downloader on "https://REDACTED.googlevideo.com/videoplayback?expire=REDACTED"
[download] Destination: Top 10 Richest Rockstars Of All Time [dlBfVitn-ow].mp4
[download]   0.9% of 39.77MiB at 608.47KiB/s ETA 01:06ERROR: I/O operation on closed file
Traceback (most recent call last):
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1382, in wrapper
    return func(self, *args, **kwargs)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1466, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1522, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 2618, in process_video_result
    self.process_info(new_info)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 3105, in process_info
    success, real_download = self.dl(temp_filename, info_dict)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 2816, in dl
    return fd.download(name, new_info, subtitle)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/downloader/common.py", line 440, in download
    ret = self.real_download(filename, info_dict)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/downloader/http.py", line 372, in real_download
    return download()
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/downloader/http.py", line 349, in download
    retry(err)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/downloader/http.py", line 236, in retry
    ctx.stream.close()
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/utils.py", line 2204, in close
    self.__exit__(self, *args, value=False, traceback=False)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/utils.py", line 2184, in __exit__
    _unlock_file(self.f)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/utils.py", line 2154, in _unlock_file
    fcntl.flock(f, fcntl.LOCK_UN)
ValueError: I/O operation on closed file
```

Resuming download:

```
# yt-dlp https://youtu.be/dlBfVitn-ow  -f 22 -v
[debug] Command-line config: ['https://youtu.be/dlBfVitn-ow', '-f', '22', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.02.04 [c1653e9ef]
[debug] Lazy loading extractors is disabled
[debug] Python version 3.10.2 (CPython 64bit) - Linux-5.15.21-1-MANJARO-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4.1 (setts), ffprobe 4.4.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, secretstorage, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: https://youtu.be/dlBfVitn-ow
[youtube] dlBfVitn-ow: Downloading webpage
[youtube] dlBfVitn-ow: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[info] dlBfVitn-ow: Downloading 1 format(s): 22
[debug] Invoking downloader on "https://REDACTED.googlevideo.com/videoplayback?expire=REDACTED"
[download] Resuming download at byte 376059


ERROR: Did not get any data blocks
  File "/home/coletdev/.local/bin/yt-dlp", line 33, in <module>
    sys.exit(load_entry_point('yt-dlp==2022.2.4', 'console_scripts', 'yt-dlp')())
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/__init__.py", line 857, in main
    _real_main(argv)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/__init__.py", line 847, in _real_main
    retcode = ydl.download(all_urls)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 3231, in download
    self.__download_wrapper(self.extract_info)(
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 3204, in wrapper
    res = func(*args, **kwargs)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1373, in extract_info
    return self.__extract_info(url, self.get_info_extractor(ie_key), download, extra_info, process)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1382, in wrapper
    return func(self, *args, **kwargs)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1466, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1522, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 2618, in process_video_result
    self.process_info(new_info)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 3105, in process_info
    success, real_download = self.dl(temp_filename, info_dict)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 2816, in dl
    return fd.download(name, new_info, subtitle)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/downloader/common.py", line 440, in download
    ret = self.real_download(filename, info_dict)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/downloader/http.py", line 372, in real_download
    return download()
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/downloader/http.py", line 341, in download
    self.report_error('Did not get any data blocks')
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/downloader/common.py", line 177, in report_error
    self.ydl.report_error(*args, **kargs)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 931, in report_error
    self.trouble(f'{self._format_err("ERROR:", self.Styles.ERROR)} {message}', *args, **kwargs)
  File "/home/coletdev/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 860, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())
```

So I think we have two issues here:
- server side issue with format 22 which has been reported before. **Currently the only solution to this is to choose another format**. Unfortunately `--check-formats` doesn't work with this. https://github.com/ytdl-org/youtube-dl/issues/27300. 
- some sort of bug in the locked file handling when downloading for the first time. We should get the same "Did not get any data blocks error"

**Comment by jakeogh** [comment]
Just found this, @coletdjnz if I am reading correctly, c1653e9efba2768910e6a5f62c88c7a95317b431 is before a3125791c7a5cdf2c8c025b99788bf686edd1a8a which might fix this? The format 22 issue may be unrelated to locking, so please test with `yt-dlp https://youtu.be/dlBfVitn-ow` as well. If that all works, https://github.com/yt-dlp/yt-dlp/pull/2656 makes advisory locking expected behavior.

**Comment by pukkandan** [comment]
The file locking issue is fixed in 747c0bd127ebd205278d31ec7216ffa02fe96734
